File: NetCoreGameServer/NetCoreGameServer/Data/Model/Group.cs
namespace NetCoreGameServer.Data.Model;

public class Group
{
    public int Id { get; set; }
    public int LeaderId { get; set; }
    public List<GroupUser> Users { get; set; }
    public GroupOptions Options { get; set; }
    public Maze Maze { get; set; }

    public Group ToListResponse()
    {
        return new Group
        {
            Id = Id,
            LeaderId = LeaderId,
            Users = Users,
            Options = Options
        };
    }
}

File: next/src/models/Group.ts
import GroupOptions from "./GroupOptions";
import GroupUser from "./GroupUser";
import { Maze } from "./Maze";

type Group = {
  id: number;
  leaderId: number;
  users: Partial<GroupUser>[];
  options: GroupOptions;
  maze?: Maze | null;
}

export default Group;